python/open3d/__init__.py

```diff
@@ -109,7 +109,14 @@ def _insert_pybind_names(skip_names=()):
     sys.modules.update(submodules)
 
 
-import open3d.visualization
+try:
```

Comment:
We should implement systematic unit tests starting from this point. 
For example, in the pytest files, expect this exception conditioned on the extra package (open3d or open3d[standard]). This can help us make sure that the core behavior is preserved without the gui requirements. Might need to tweak some of the CIs, but a summary of some local tests would be also helpful.

Reply:
This may not be quite critical for this PR, but if we plan on more advanced dependency management I think it would be required.